numba/cpython/listobj.py

```diff
@@ -921,57 +935,49 @@ def list_extend(lst, iterable):
 
     return context.compile_internal(builder, list_extend, sig, args)
 
-@lower_builtin("list.index", types.List, types.Any)
-def list_index(context, builder, sig, args):
-
-    def list_index_impl(lst, value):
-        for i in range(len(lst)):
-            if lst[i] == value:
-                return i
-        # XXX references are leaked when raising
-        raise ValueError("value not in list")
-
-    return context.compile_internal(builder, list_index_impl, sig, args)
-
-@lower_builtin("list.index", types.List, types.Any,
-           types.Integer)
-def list_index(context, builder, sig, args):
 
-    def list_index_impl(lst, value, start):
-        n = len(lst)
-        if start < 0:
-            start += n
+@overload_method(types.List, "index")
+def list_index(lst, value, start=None, end=None):
```

Comment:
This isn't the signature for `list.index`? It's, e.g.:
```pycon
In [1]: list.index?
Signature: list.index(self, value, start=0, stop=9223372036854775807, /)
```
I suspect the `stop` is `max(intp)`.

Reply:
Done!